Brokerages/Samco/SamcoBrokerage.cs

```diff
@@ -0,0 +1,1009 @@
+                }
+
+                var message = $"Order failed, Order Id: {order.Id} timestamp: {order.Time} quantity: {order.Quantity} content: {orderResponse.statusMessage}";
+                OnOrderEvent(new OrderEvent(order, DateTime.UtcNow, orderFee, "Samco Order Event") { Status = OrderStatus.Invalid });
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, -1, message));
+
+                submitted = true;
+                return;
+            });
+            return submitted;
+        }
+
+        private OrderStatus ConvertOrderStatus(OrderDetails orderDetails)
+        {
+            var filledQty = Convert.ToInt32(orderDetails.filledQuantity, CultureInfo.InvariantCulture);
+            var pendingQty = Convert.ToInt32(orderDetails.pendingQuantity, CultureInfo.InvariantCulture);
+            var orderDetail = _samcoAPI.GetOrderDetails(orderDetails.orderNumber);
+            if (orderDetails.orderStatus != "complete" && filledQty == 0)
+            {
+                return OrderStatus.Submitted;
+            }
+            else if (filledQty > 0 && pendingQty > 0)
+            {
+                return OrderStatus.PartiallyFilled;
+            }
+            else if (pendingQty == 0)
+            {
+                return OrderStatus.Filled;
+            }
+            else if (orderDetail.orderStatus.ToUpperInvariant() == "CANCELLED")
+            {
+                return OrderStatus.Canceled;
+            }
+
+            return OrderStatus.None;
+        }
+
+        private void EmitFillOrder(SamcoOrderResponse orderResponse)
+        {
+            try
+            {
+                var brokerId = orderResponse.orderNumber;
+                var orderDetails = orderResponse.orderDetails;
+                var order = CachedOrderIDs
+                    .FirstOrDefault(o => o.Value.BrokerId.Contains(brokerId))
+                    .Value;
+                if (order == null)
+                {
+                    order = _algorithm.Transactions.GetOrderByBrokerageId(brokerId);
+                    if (order == null)
+                    {
+                        // not our order, nothing else to do here
+                        return;
+                    }
+                }
+
+                var brokerageSecurityType = _symbolMapper.GetBrokerageSecurityType(orderDetails.tradingSymbol);
+                var symbol = _symbolMapper.GetLeanSymbol(orderDetails.tradingSymbol, brokerageSecurityType);
+                var fillPrice = decimal.Parse(orderDetails.filledPrice, NumberStyles.Float, CultureInfo.InvariantCulture);
+                var fillQuantity = decimal.Parse(orderDetails.filledQuantity, NumberStyles.Float, CultureInfo.InvariantCulture);
+                var updTime = DateTime.UtcNow;
+                var security = _securityProvider.GetSecurity(order.Symbol);
+                var orderFee = security.FeeModel.GetOrderFee(new OrderFeeParameters(security, order));
```

Comment:
Fees should be on the filled quantity only really, else for partial fills will apply multiple times the same fees